src/events/tab_event_handler.ts

```diff
@@ -0,0 +1,82 @@
+import yaml from "js-yaml";
+import vscode from 'vscode';
+import path from 'path';
+import fs from 'fs';
+
+import { VSCODE } from "../constants";
+import { ITab, ITabFile } from "../interface";
+import { formatDatetime, getBranch } from "../utils/common";
+import { ConfigUtils } from "../utils/config_utils";
+import { pathUtils } from "../utils/path_utils";
+import { generateSettings } from "../settings";
+import { RepoState } from "../utils/repo_state_utils";
+import { UserState } from "../utils/user_utils";
+
+export class tabEventHandler {
+	repoPath = "";
+	branch = "";
+	pathUtils: any;
+
+	// Diff props
+	settings = generateSettings();
+	shouldProceed = false;
+
+	constructor(repoPath="") {
+		const userState = new UserState();
+		const isValidAccount = userState.isValidAccount();
+		this.repoPath = repoPath || pathUtils.getRootPath();
+		if(!this.repoPath) return;
+		const repoState = new RepoState(this.repoPath).get();
+		const repoIsConnected = repoState.IS_CONNECTED;
+		this.shouldProceed = isValidAccount && repoIsConnected;
+		if (!this.shouldProceed) return;
+		this.branch = getBranch(this.repoPath);
+		this.pathUtils = new pathUtils(this.repoPath, this.branch);
+	}
+
+	handleTabChangeEvent = (isTabEvent: boolean = true) => {
+		if(!this.repoPath) return;
```

Comment:
can combin both in single statement 
if(!this.repoPath o|| !isTabEvent) return;